app/renderer/components/navigation/urlBar.js

```diff
@@ -357,10 +357,13 @@ class UrlBar extends React.Component {
   componentDidUpdate (prevProps) {
     // this.urlInput is not initialized in titleMode
     if (this.urlInput) {
+      const pdfjsEnabled = getSetting(settings.PDFJS_ENABLED)
       if (this.props.activeFrameKey !== prevProps.activeFrameKey) {
         this.keyPressed = false
         // The user just changed tabs
-        this.setValue(this.props.locationValue)
+        this.setValue(this.props.locationValue !== 'about:blank'
+          ? this.props.locationValue
+          : UrlUtil.getDisplayLocation(this.props.location, pdfjsEnabled))
```

Comment:
Since this doesn't do anything, I'm changing this to:
```
-        this.setValue(this.props.locationValue !== 'about:blank'
-          ? this.props.locationValue
-          : UrlUtil.getDisplayLocation(this.props.location, pdfjsEnabled))
+        if (this.props.locationValue !== 'about:blank') {
+          this.setValue(UrlUtil.getDisplayLocation(this.props.location, pdfjsEnabled))
+        }
```

which I think is the expected functionality.